modules/opencensus/src/main/java/org/apache/ignite/internal/processors/monitoring/opencensus/OpenCensusMetricExporterSpi.java

```diff
@@ -0,0 +1,279 @@
+import org.apache.ignite.spi.metric.BooleanMetric;
+import org.apache.ignite.spi.metric.DoubleMetric;
+import org.apache.ignite.spi.metric.IntMetric;
+import org.apache.ignite.spi.metric.LongMetric;
+import org.apache.ignite.spi.metric.Metric;
+import org.apache.ignite.internal.processors.metric.MetricRegistry;
+import org.apache.ignite.spi.metric.ObjectMetric;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * <a href="https://opencensus.io">OpenCensus</a> monitoring exporter. <br>
+ * <br>
+ * This class will export all Ignite metrics with the OpenCensus API.<br>
+ * <br>
+ * Please, note, metrics recorded with the OpenCensus API each {@link #period} milliseconds.
+ * <br>
+ * To enable export from OpenCensus to the wild user should configure OpenCensus exporter.
+ * Please, see <a href="https://opencensus.io/exporters/supported-exporters/java/">OpenCensus documentation</a> for additional information.
+ *
+ * Example of exporter configuration:
+ * <pre>
+ * {@code
+ *   PrometheusStatsCollector.createAndRegister();
+ *
+ *   HTTPServer server = new HTTPServer("localhost", 8888, true);
+ * }
+ * </pre>
+ *
+ * @see MetricRegistry
+ */
+public class OpenCensusMetricExporterSpi extends PushMetricsExporterAdapter {
+    /** Flag to enable or disable tag with Ignite instance name. */
+    private boolean sendInstanceName;
+
+    /** Flag to enable or disable tag with Node id. */
+    private boolean sendNodeId;
+
+    /** Flag to enable or disable tag with Consistent id. */
+    private boolean sendConsistentId;
+
+    /** Ignite instance name. */
+    public static final TagKey INSTANCE_NAME_TAG = TagKey.create("iin");
+
+    /** Ignite node id. */
+    public static final TagKey NODE_ID_TAG = TagKey.create("ini");
+
+    /** Ignite node consistent id. */
+    public static final TagKey CONSISTENT_ID_TAG = TagKey.create("inci");
+
+    /** Ignite instance name in the form of {@link TagValue}. */
+    private TagValue instanceNameValue;
+
+    /** Ignite node id in the form of {@link TagValue}. */
+    private TagValue nodeIdValue;
+
+    /** Ignite consistent id in the form of {@link TagValue}. */
+    private TagValue consistenIdValue;
+
+    /**
+     * Tags that will be exported with each measure
+     *
+     * @see #sendInstanceName
+     * @see #sendNodeId
+     */
+    private List<TagKey> tags = new ArrayList<>();
+
+    /**
+     * Opencensus measures.
+     * Values obtained from Ignite recorded to them.
+     */
+    private Map<String, Measure> measures = new HashMap<>();
+
+    /** */
+    private static Function<Metric, Measure> CREATE_LONG = m ->
+        MeasureLong.create(m.name(), m.description(), "");
+
+    /** */
+    private static Function<Metric, Measure> CREATE_DOUBLE = m ->
+        MeasureDouble.create(m.name(), m.description(), "");
+
+    /** {@inheritDoc} */
+    @Override public void export() {
+        StatsRecorder recorder = Stats.getStatsRecorder();
+
+        try (Scope globalScope = tagScope()) {
```

Comment:
'globalScope' is never used

Reply:
It's OK. 

Scope used by the undelying structures in `MeasureMap#put` calls.